cranelift/codegen/src/isa/x64/inst.isle

```diff
@@ -86,6 +86,12 @@
               (dst_lo WritableGpr)
               (dst_hi WritableGpr))
 
+       ;; x64 'mul' instruction but it only outputs the low half
```

Comment:
Is there any reason we can't use the ordinary `mul` in the `AluRmiR`? Alternately, if we do this, we should remove it completely from `AluRmiROpcode` and switch uses over -- there shouldn't be two different ways to build a MachInst for it.

Reply:
`AluRmiROpcode.Mul` is a signed multiply, it emits `imul`. This one emits `mul`, which is unsigned multiplication. I added it as a new MachInst since the implementation for the `AluRmiR` mul operation states that it is freeloaded into the MachInst and doesn't really fit it already so I didn't add a `AluRmiROpcode.UMul`.

Reply:
Ah! I had missed that difference; thanks. This is indeed fine then.